plugins/destination/postgresql/client/list_tables.go

```diff
@@ -22,7 +22,11 @@ SELECT
 	columns.ordinal_position AS ordinal_position,
 	pg_class.relname AS table_name,
 	pg_attribute.attname AS column_name,
-	pg_catalog.format_type(pg_attribute.atttypid, pg_attribute.atttypmod) AS data_type,
+	CASE
+	    WHEN pg_type.typcategory = 'A' AND pg_type.typelem != 0
+		THEN pg_catalog.format_type(pg_type.typelem, pg_attribute.atttypmod) || '[]'
+		ELSE pg_catalog.format_type(pg_attribute.atttypid, pg_attribute.atttypmod)
+	END AS data_type,
```

Comment:
What's the reasoning behind the change here?

Reply:
It's the point 2 in the description.

There's a difference in `pg_catalog.format_type` implementations between CocroachDB & PostgreSQL.

The `list<uint64>` Apache Arrow type is stored as `numeric(20,0)[]` in both databases, but then `pg_catalog.format_type(pg_attribute.atttypid, pg_attribute.atttypmod)` would return:
* `numeric(20,0)[]` in PostgreSQL
* `numeric[]` in CockroachDB

This query change allows us to have `numeric(20,0)[]` returned from both databases

Reply:
I see, thanks for the explanation! Could we maybe add a comment to the query here to explain that this is for CockroachDB compatibility? I'm thinking that we may want to split CockroachDB into its own plugin down the line because of all the subtle differences, so knowing what is for Cockroach will be useful then

Reply:
Added a comment in 13cd014